model/riscv_fdext_regs.sail

```diff
@@ -265,8 +265,8 @@ function rF_or_X_D(i) = {
   else if xlen >= 64
   then X(i)[63..0]
   else {
-    assert(i[0] == bitzero);
-    if i == zeros() then zeros() else X(i + 1) @ X(i)
+    assert((let Regidx(i_bits) = i in i_bits)[0] == bitzero);
+    if i == zreg then zeros() else X(regidx_offset(i, 0b00001)) @ X(i)
```

Comment:
Similarly, we have some operator overloads for `physaddr`:

```
function sub_virtaddr_xlenbits(virtaddr(addr) : virtaddr, offset : xlenbits) -> virtaddr = virtaddr(addr - offset)

overload operator - = { sub_virtaddr_xlenbits }
```

Reply:
If you'd prefer to eliminate the `regidx_offset` in favor of an overload, can I ask that it be done in a later PR?

Reply:
I agree it makes sense to do that in a follow-up PR